### PR TITLE
Remove State Sync execution step

### PR DESCRIPTION
**Pull request by vcastellm**
# Description

Do not execute State Syncs internally from the client

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

E2E tests are not expected to pass until merging sub-sequent PRs

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Review by goran-ethernal — APPROVED** [review]

**Review by lazartravica — APPROVED** [review]

File: consensus/polybft/consensus_runtime.go [hunk 1]
```diff
@@ -309,21 +309,6 @@ func (c *consensusRuntime) populateFsmIfBridgeEnabled(
 		if err := c.state.cleanCommitments(nextStateSyncExecutionIdx); err != nil {
 			return fmt.Errorf("cannot clean commitments: %w", err)
 		}
-
-		nonExecutedCommitments, err := c.state.getNonExecutedCommitments(nextStateSyncExecutionIdx)
```
Comment (Stefan-Ethernal) [inline consensus/polybft/consensus_runtime.go:313]: Should we remove `func (s *State) getNonExecutedCommitments(startIndex uint64) ([]*CommitmentMessageSigned, error)` altogether? It is being used only in the unit test currently.

File: consensus/polybft/consensus_runtime.go [hunk 2]
```diff
@@ -309,21 +309,6 @@ func (c *consensusRuntime) populateFsmIfBridgeEnabled(
...
-			return fmt.Errorf("cannot get non executed commitments: %w", err)
-		}
-
-		if len(nonExecutedCommitments) > 0 {
-			bundlesToExecute, err := c.state.getBundles(nextStateSyncExecutionIdx, maxBundlesPerSprint)
-			if err != nil {
-				return fmt.Errorf("cannot get bundles: %w", err)
-			}
-
-			ff.commitmentsToVerifyBundles = nonExecutedCommitments
```
Comment (Stefan-Ethernal) [inline consensus/polybft/consensus_runtime.go:324]: Was it safe to remove this? Are we removing:
1. just bundles execution from client side or 
2. bundle verification as part of the consensus as well?
If it is 2., then I'd say we can easily remove `case *BundleProof` from the `fsm.VerifyStateTransactions` function. WDYT?

File: consensus/polybft/fsm.go [hunk 3]
```diff
@@ -203,24 +203,6 @@ func (f *fsm) stateTransactions() []*types.Transaction {
 		}
 	}
 
-	if f.isEndOfSprint {
-		for _, bundle := range f.bundleProofs {
-			inputData, err := bundle.EncodeAbi()
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm.go:208]: It seems we don't need `EncodeAbi` (possibly even `DecodeAbi`, needs checking) anymore? Therefore we don't need to make `BundleProof` implement `StateTransactionInput` interface anymore...?

Comment (vcastellm) [reply consensus/polybft/fsm.go:208]: Starting just after this PR is merged (we need this for the release), I'll start removing everything related to bundles, I don't want to introduce more that just the necessary in this PR, makes sense?

File: consensus/polybft/consensus_runtime.go (hunk 2 above)
Comment (vcastellm) [reply consensus/polybft/consensus_runtime.go:324]: It's 1, just bundle execution, we still maintain bundle verification in consensus. As commented in https://github.com/0xPolygon/polygon-edge/pull/964#discussion_r1033858701 removal of all bundle related things to come.

File: consensus/polybft/consensus_runtime.go (hunk 1 above)
Comment (vcastellm) [reply consensus/polybft/consensus_runtime.go:313]: Yes we can safely remove execution related things https://github.com/0xPolygon/polygon-edge/pull/964/commits/8a01208de461ddfdc7e5ac166096b505aba05e3b

**Review by b00ris — APPROVED** [review]

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**